proto/process/agent.proto

```diff
@@ -566,7 +567,7 @@ message Connection {
 	uint32 lastTcpClosed = 31;
 
 	// dns stats based on domain queried, the key corresponds to an index into the `domains` field
-	map<int32, DNSStats> dnsStatsByDomain = 34;
+	map<int32, DNSStatsByQueryType> dnsStatsByDomain = 42;
```

Comment:
Instead of replacing, let's add a new field. Otherwise, any backend app using the this payload version won't be able to handle data from older agents.